cds/modules/deposit/api.py

```diff
@@ -988,26 +988,29 @@ def edit(self, pid=None):
         assert video_new.report_number
         return video_new
 
-    def _delete_flows(self, keep_last=True):
-        """Delete the flows."""
-        if keep_last:
-            flows = FlowMetadata.get_by_deposit(
-                deposit_id=self["_deposit"]["id"], is_last=False, multiple=True
+    def _delete_flows(self, hard=False):
+        """Delete the flows.
+
+        :param hard: If True delete all flows, if false keep last one.
+        """
+        if hard:
+            flows = FlowMetadata.get_all_by_deposit(
+                deposit_id=self["_deposit"]["id"]
             )
         else:
             flows = FlowMetadata.get_by_deposit(
-                deposit_id=self["_deposit"]["id"], multiple=True
+                deposit_id=self["_deposit"]["id"], is_last=False, multiple=True
```

Comment:
I think we should not delete flow on soft deletion, not even the previous ones (not real benefit?), we should simply stop everything.
So I would change the code to:
```
flows = FlowMetadata.get_all_by_deposit(
    deposit_id=self["_deposit"]["id"]
)
for flow in flows:
    if hard:
        FlowService(flow).delete()
    else:
        FlowService(flow).cancel()
```